imports/plugins/core/files/server/fileCollections.js

```diff
@@ -24,7 +24,7 @@ async function lazyLoadSharp() {
 }
 
 FileRecord.downloadEndpointPrefix = "/assets/files";
-FileRecord.absoluteUrlPrefix = Meteor.absoluteUrl();
+FileRecord.absoluteUrlPrefix = Meteor.absoluteUrl(); // TODO: Reaction.absoluteUrl()
```

Comment:
this line was added after the PR was created and I haven't had time to test whether the change to `Reaction.absoluteUrl` would work in this context. I'm confident in the change at runtime but haven't investigated initialization time yet.

Reply:
@aldeed maybe you can help me here. adding an import statement results in an undefined `Reaction`, which I presume means this code runs before we build `Reaction`. Should it? I'm curious whether this code should live inside a function, and kicked off in a startup script once all code has been loaded.

Reply:
@pmn4 `Reaction` obj should never be undefined when imported. Are you sure the import was correct? `import { Reaction } from "/server/api";`

Reply:
@aldeed I haven't dug in yet, but trying to start the app yields:
``` 
W20180425-13:38:34.514(-4)? (STDERR) TypeError: Cannot read property 'absoluteUrl' of undefined
W20180425-13:38:34.514(-4)? (STDERR)     at fileCollections.js (imports/plugins/core/files/server/fileCollections.js:27:41)
W20180425-13:38:34.514(-4)? (STDERR)     at fileEvaluate (packages/modules-runtime.js:343:9)
W20180425-13:38:34.514(-4)? (STDERR)     at require (packages/modules-runtime.js:238:16)
W20180425-13:38:34.515(-4)? (STDERR)     at index.js (imports/plugins/core/files/server/index.js:1:57)
W20180425-13:38:34.515(-4)? (STDERR)     at fileEvaluate (packages/modules-runtime.js:343:9)
W20180425-13:38:34.515(-4)? (STDERR)     at require (packages/modules-runtime.js:238:16)
W20180425-13:38:34.515(-4)? (STDERR)     at email.js (server/api/core/email/email.js:1:567)
W20180425-13:38:34.516(-4)? (STDERR)     at fileEvaluate (packages/modules-runtime.js:343:9)
W20180425-13:38:34.516(-4)? (STDERR)     at require (packages/modules-runtime.js:238:16)
W20180425-13:38:34.516(-4)? (STDERR)     at index.js (server/api/core/email/index.js:1:78)
W20180425-13:38:34.516(-4)? (STDERR)     at fileEvaluate (packages/modules-runtime.js:343:9)
W20180425-13:38:34.517(-4)? (STDERR)     at require (packages/modules-runtime.js:238:16)
W20180425-13:38:34.517(-4)? (STDERR)     at index.js (server/api/core/index.js:1:334)
W20180425-13:38:34.517(-4)? (STDERR)     at fileEvaluate (packages/modules-runtime.js:343:9)
W20180425-13:38:34.517(-4)? (STDERR)     at require (packages/modules-runtime.js:238:16)
W20180425-13:38:34.518(-4)? (STDERR)     at index.js (server/api/index.js:1:190)
W20180425-13:38:34.518(-4)? (STDERR)     at fileEvaluate (packages/modules-runtime.js:343:9)
W20180425-13:38:34.518(-4)? (STDERR)     at require (packages/modules-runtime.js:238:16)
W20180425-13:38:34.518(-4)? (STDERR)     at core.js (lib/api/core.js:10:10)
W20180425-13:38:34.519(-4)? (STDERR)     at fileEvaluate (packages/modules-runtime.js:343:9)
W20180425-13:38:34.519(-4)? (STDERR)     at require (packages/modules-runtime.js:238:16)
W20180425-13:38:34.519(-4)? (STDERR)     at index.js (lib/api/index.js:1:313)
```

Reply:
there is a line already which imports `Logger` from "/server/api". I added Reaction to that statement:
``` JavaScript
import { Logger, Reaction } from "/server/api";
console.log("Immediate", !!Logger, !!Reaction);
setTimeout(() => console.log("Later", !!Logger, !!Reaction), 5000);
```
the console log yields
```
Immediate false false
// ... some other console messages ...
Later true true
```

This behavior is consistent with a circular dependency.
I suspect it was never identified previously because the depencies are eventually resolved. This is likely one of the few places where we import code and try to use it immediately.

Reply:
@pmn4 Are you able to identify where the circular dependency is and unravel it? Or for example could you import your new domains object directly rather than the whole `Reaction` object?